tools/clang/lib/Frontend/ASTUnit.cpp

```diff
@@ -2054,14 +2054,10 @@ ASTUnit *ASTUnit::LoadFromCommandLine(
   AST.reset(new ASTUnit(false));
   // HLSL Change Starts
   AST->HlslLangExtensions = HlslLangExtensions;
-  // Enable -verify on the libclang initialization path.
-  bool VerifyDiagnostics = false;
-  for (const char** Arg = ArgBegin; Arg != ArgEnd; ++Arg) {
-    if (strcmp(*Arg, "-verify") == 0) {
-      VerifyDiagnostics = true;
-      break;
-    }
-  }
+  // Enable -verify and -verify-ignore-unexpected on the libclang initialization path.
+  bool VerifyDiagnostics = CI->getDiagnosticOpts().VerifyDiagnostics;
+  Diags->getDiagnosticOptions().setVerifyIgnoreUnexpected(
+	  CI->getDiagnosticOpts().getVerifyIgnoreUnexpected());
```

Comment:
@tex3d Do you know this code? There are two `DiagnosticsOpts` object, one that lives in the `CompilationInstance` and gets parsed from the command line, and one that was created by the callee that apparently gets used. This is how I was able to use `-verify-ignore-unexpected=note` in my front-end test.